### PR TITLE
Create phone standardization wrapper with centralized util functions

### PR DESCRIPTION
**Pull request by BradySkylight**
### Description
Create GCP Cloud Function wrapper for the SDK standardize_phones to ensure that all phone numbers have the same format in the FHIR Resources/Bundles passed into the function.

In addition to the GCP Cloud Function wrapper being created the following other changes were made:

Centralize command validation functions into a single location (started with utils.py, but moving to SDK PHDI Project)
Modify other cloud functions (upload_fhir_bundle, http_standardize_names) to use the new centralized util functions
Added more data into the assets single_patient_bundle to be used for testing in the new function
Added tests for new standardize_phones function

### CODE REVIEW

File: cloud-functions/utils.py
Comment (nickclyde) [inline cloud-functions/utils.py:7]: I think you're right here - ideally the cloud functions code is as minimal as possible, pulling any shared code from the SDK. Plus the way that we will bundle these functions w/ terraform means that they won't be able to reference the directory above themselves, so it probably makes the most sense to slide this over to the other repo.

File: cloud-functions/http_standardize_phones/main.py
Comment (nickclyde) [inline cloud-functions/http_standardize_phones/main.py:38]: 🎉

**Review by nickclyde — CHANGES_REQUESTED** [review]
Looks great! But I agree that the utils.py code belongs in the SDK

Also, small nit: Be sure to make the PR title more descriptive, since when we squash merge, the PR title is what shows up in the commit log on `main`

File: cloud-functions/utils.py
Comment (BradySkylight) [reply cloud-functions/utils.py:7]: Yes I think having this moved over to the SDK project makes the most sense and will clean up the GCP Cloud Function code a great deal.  Maybe we need to open a new ticket for us to handle this.

File: cloud-functions/http_standardize_names/main.py
Comment (DanPaseltiner) [inline cloud-functions/http_standardize_names/main.py:36]: Are we being inconsistent here with our return type? Looks like we always return a `flask.Response` object except when everything works properly. I believe `final_response` is a `dict`.

File: cloud-functions/http_standardize_names/requirements.txt
Comment (DanPaseltiner) [inline cloud-functions/http_standardize_names/requirements.txt:32]: I believe this can be changed to simply `phdi` now that it is available on pypi. Unless we need something on the `main` branch of `phdi` that is not yet available in the PyPi release I think we should start transitioning away from installing phdi from GitHub

**Review by DanPaseltiner — APPROVED** [review]

**Review by DanPaseltiner — CHANGES_REQUESTED** [review]
Oops, looks like I hit approve before when I meant to request changes.

File: cloud-functions/http_standardize_names/main.py
Comment (BradySkylight) [reply cloud-functions/http_standardize_names/main.py:36]: I was thinking about this.  Should we construct a flask.Response that contains the json result that is passed back in?  That's where my head is at for this.

Comment (DanPaseltiner) [reply cloud-functions/http_standardize_names/main.py:36]: That's what I was thinking.

**Review by nickclyde — APPROVED** [review]
Awesome work on this Brady!!

File: cloud-functions/http_standardize_names/main.py
Comment (DanPaseltiner) [inline cloud-functions/http_standardize_names/main.py:33]: nit: I believe a leading underscore indicates an internal method in Python. Importing this outside of the utils module seems to suggest that it isn't just an internal method. Should we consider renaming? Also, my preference would be for the function name to be a little more descriptive to make the purpose clearer to users. Maybe something like `make_json_content_response`? I'm just not quite sure that `_full_response` makes it clear what this function does.

File: cloud-functions/http_standardize_phones/main.py
Comment (DanPaseltiner) [inline cloud-functions/http_standardize_phones/main.py:41]: Are these some debugging artifacts?

File: cloud-functions/upload_fhir_bundle/main.py
Comment (DanPaseltiner) [inline cloud-functions/upload_fhir_bundle/main.py:69]: I think we might have a type inconsistency here. Looks like the `message` argument of `_fail` is intended to be a string, but `error_as_dictionary` is a dict. Maybe we should just drop line 64 and pass `error` directly to `_fail`.

**Review by DanPaseltiner — COMMENTED** [review]
Thanks @BradySkylight for continuing to push forward on this! I have included a few more comments. I feel that the second 2 are blocking.

File: cloud-functions/http_standardize_names/main.py
Comment (BradySkylight) [reply cloud-functions/http_standardize_names/main.py:33]: I have fixed this in another PR specific to the utils package.  I was thinking about this last night and my plan was to get rid of full_response all together and just use either success or fail as responses.  I think this is a cleaner method.  Thoughts?

File: cloud-functions/http_standardize_phones/main.py
Comment (BradySkylight) [reply cloud-functions/http_standardize_phones/main.py:41]: Yes, removing them.  Thanks!

File: cloud-functions/upload_fhir_bundle/main.py
Comment (BradySkylight) [reply cloud-functions/upload_fhir_bundle/main.py:69]: Sounds good.  Will fix this.

File: cloud-functions/http_standardize_names/main.py
Comment (DanPaseltiner) [reply cloud-functions/http_standardize_names/main.py:33]: Yep, that works for me.

File: cloud-functions/upload_fhir_bundle/main.py
Comment (DanPaseltiner) [reply cloud-functions/upload_fhir_bundle/main.py]: Oops actually looks like `error.json()` is a string. `error` is an `ValidationError` object.

File: cloud-functions/http_standardize_names/main.py
Comment (DanPaseltiner) [inline cloud-functions/http_standardize_names/main.py:33]: PR 37 currently replaces `success` with `make_response`.

File: cloud-functions/http_standardize_phones/main.py
Comment (DanPaseltiner) [inline cloud-functions/http_standardize_phones/main.py:38]: PR 37 currently replaces success with make_response.

File: cloud-functions/upload_fhir_bundle/main.py
Comment (DanPaseltiner) [inline cloud-functions/upload_fhir_bundle/main.py:63]: Can replace with `log_error_and_generate_response`.

File: cloud-functions/upload_fhir_bundle/main.py
Comment (DanPaseltiner) [inline cloud-functions/upload_fhir_bundle/main.py:88]: Only realizing as I do this review that `response` is a `dict` not a `flask.Response` object. This is a mistake I made when originally writing this  function. I think it was before we moved to returning `flask.Responses`

File: cloud-functions/upload_fhir_bundle/test_upload_fhir_bundle.py
Comment (DanPaseltiner) [inline cloud-functions/upload_fhir_bundle/test_upload_fhir_bundle.py:46]: PR 37 currently replaces success with make_response.

File: cloud-functions/upload_fhir_bundle/test_upload_fhir_bundle.py
Comment (DanPaseltiner) [inline cloud-functions/upload_fhir_bundle/test_upload_fhir_bundle.py:67]: PR 37 currently replaces success with make_response, and `status` is replaced with `status_code.

**Review by DanPaseltiner — APPROVED** [review]
Brady, this looks awesome! There are a couple things we'll need to realign with `main` after PR 37 lands, but I don't want a review from me to be blocking. Thanks for your work on this.

**Review by nickclyde — APPROVED** [review]
:shipit:

File: cloud-functions/upload_fhir_bundle/main.py
Comment (BradySkylight) [reply cloud-functions/upload_fhir_bundle/main.py:88]: I think this is ok here though.  It seems to still function like a flask.response object.  And we are only interested if the upload was successful or not.  I can do some reconstructing here if we need to, but maybe in a different PR/Branch though.  ;)

**Pull request merged**
